sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -177,12 +180,52 @@ class NoOpStatementExec extends LeafStatementExec {
  *   Executable nodes for nested statements within the CompoundBody.
  * @param label
  *   Label set by user to CompoundBody or None otherwise.
+ * @param isScope
+ *   Flag that indicates whether Compound Body
```

Comment:
not clear what does it mean. so, `CompoundBodyExec(isScope = false)` is not a compound body?

Reply:
It is Compound Body but it is not scope.  Comment was not completed, I updated it now.

body1 is compound body but it is not a scope if it doesn't have `BEGIN` and `END`.
```
IF condition THEN
    body1;
END IF;
```

This compound body is scope (label is either defined by user or implicitly.
```
BEGIN
    stmt1;
    stmt2;
    ...
END
```